scenes/cslib_object.txt

```diff
@@ -0,0 +1,111 @@
+*comment *create dragon_fieldname_max	2
+*comment
+*comment *create dragon_name		""
+*comment *create dragon_hp			""
+*comment
+*comment *create dragon_1_name 		"Wally"
+*comment *create dragon_1_hp 		100
+*comment *create dragon_2_name 		"Dave"
+*comment *create dragon_2_hp		150
+*comment
+*comment EXAMPLE of getting a dragon instance
+*comment
+*comment *gosub_scene cslib_object get "dragon" 2
+*comment 	sets dragon_name to "Dave" and dragon_hp to 150 and dragon_index to 2
+*comment
+*comment EXAMPLE of setting a dragon instance
+*comment
+*comment *gosub_scene cslib_object set "dragon" 2 "name" "Sir David"
+*comment    sets dragon_2_name to "Sir David"
+*comment	now when you get dragon 2, dragon_name will be "Sir David"
+*comment	this will persist until you set it again
+
+*comment GET
+*comment ------------------------------------------------------------------
+*comment Takes the instance of a object specified by the index and copies
+*comment the data into the main instance
+*comment ------------------------------------------------------------------
+*comment
+*comment params:
+*comment	p_objectname (string): the name of the object
+*comment	p_index (number): index of this object instance
+*comment returns:
+*comment	none
+*comment usage example:
+*comment 	cslib_object get "dragon" 2
+*comment ------------------------------------------------------------------
+
+*label get
+*params p_objectname p_index
+*set cslib_ret ""
+*gosub _set_main_instance p_objectname p_index
+*return
+
+*comment SET
+*comment ------------------------------------------------------------------
+*comment Set the value of a particular field in a particular instance of a object
+*comment ------------------------------------------------------------------
+*comment params:
+*comment	p_objectname (string): the name of the object
+*comment	p_index (number): index of this object instance
+*comment	p_fieldname (string): the field to update
+*comment	p_value (string): the new value
+*comment returns:
+*comment	none
+*comment usage example:
+*comment 	cslib_object set "dragon" 2 "name" "Bruce"
+*comment ------------------------------------------------------------------
+
+*label set
+*params p_objectname p_index p_fieldname p_value
+*set cslib_ret ""
+*set {p_objectname&"_${p_index}_${p_fieldname}"} p_value
+*return
+
+*comment ------------------------------------------------------------------
+*comment --- PRIVATE METHODS
+*comment ------------------------------------------------------------------
+
+*comment --- ------------------------------------------------------------------
+*comment --- Copies the values in the instance at the given index into the main instance
+*comment --- ------------------------------------------------------------------
+*label _set_main_instance
+*params p_object p_index
+*temp n 1
+*temp field_count {p_object&"_fieldname_max"}
+*label _set_main_loop
+*if (n <= field_count)
+	*temp field {p_object&"_fieldname_${n}"}
```

Comment:
indentation?

Reply:
I am setting *temp field within the loop so I need the indentation.

Arguably, depending on the compiler implementation, it could be more efficient to have it declared outside the loop.

I don't have objection to moving it.

Reply:
Moved

Reply:
I think Diego was probably commenting on the type of indentation, rather than the code or presence of indentation.
E.g. you've used a tab here, I think? Would you prefer spaces @diegoboston ?

I don't mind as, to my embarrassment, we haven't been consistent across other files either (cslib_string uses tabs, cslib_array is 4 spaces). I'd suggest keeping as-is for now. I can make a note to normalise indentation across the board somewhere down the line.

Reply:
Ah, yes. When it comes to the great religious war I've always sided with the tabs. As you say, it can be normalised later (and probably there should be a note in the style guide)

Reply:
Agreed, resolving.